byterun/str.c

```diff
@@ -31,6 +31,22 @@ CAMLexport mlsize_t caml_string_length(value s)
   return temp - Byte (s, temp);
 }
 
+CAMLexport int caml_string_is_c_safe (value s)
+{
+  mlsize_t len;
+  char *p;
+  int i;
+
+  len = caml_string_length(s);
+  p = String_val(s);
+
+  for (i=0; i<len; ++i) {
```

Comment:
As I indicated in the bugreport discussion, I think `strlen` should be used instead of an explicit loop. Most C libraries have optimized versions of `strlen` (see eg. [How does the glibc strlen implementation works?](http://stackoverflow.com/questions/20021066/how-the-glibc-strlen-implementation-works) which may perform better than what the compiler can generate from the direct loop.